packages/app/components/Settings/EditCourse.tsx

```diff
@@ -12,33 +12,79 @@ const CourseRosterPageComponent = styled.div`
 `;
 
 export default function EditCourse({
-  courseId
+  courseId,
 }: CourseRosterPageProps): ReactElement {
-  const cid=courseId;
-  const course=useCourse(cid);
+  const cid = courseId;
+  const course = useCourse(cid);
+  const inviteCode = course.course?.courseInviteCode;
+  const id = course.course?.id;
+
+  const baseURL =
+    typeof window !== "undefined"
+      ? `${window.location.protocol}//${window.location.host}`
+      : "";
+
   const [form] = Form.useForm();
-  const submit=async () => {
+  const [copied, setCopied] = useState(false);
+  const [courseCode, setCourseCode] = useState(course.course?.courseInviteCode);
+  const [inviteURL, setInviteURL] = useState<string>("");
+
+  useEffect(() => {
+    setInviteURL(`${baseURL}/course/${id}/invite?code=${courseCode}`);
+  }, [courseCode, id, baseURL]);
+
+  const handleCopy = () => {
+    navigator.clipboard.writeText(inviteURL).then(() => {
+      setCopied(true);
+    });
+  };
+
+  const submit = async (values) => {
     const value = await form.validateFields();
-    await API.course.editCourseInfo(cid,{courseId:cid, questionTimer:value.questionTimer}).then(()=>message.success("Edited Course info"))
-  }
-    return (
+    await API.course
+      .editCourseInfo(cid, {
+        courseId: cid,
+        courseInviteCode: value.inviteCode,
+      })
+      .then(() => message.success("Edited Course info"));
+
+    setCopied(false);
+    setCourseCode(values.inviteCode);
+  };
+
+  return (
     <div>
       <CourseRosterPageComponent>
         <div style={{ textAlign: "center" }}>
-        <Form 
-          form={form} 
-          initialValues={course}
-          onFinish={submit}>
-            <h3> Current Timer: {course.course?.questionTimer} minutes</h3> 
-        <Form.Item style={{ width:"500px" }}label="Question Timer" name="questionTimer">
-          <Input allowClear={true} placeholder={"Minutes before a question times out"}/>
-        </Form.Item>
-        <Form.Item >
-      <Button type="primary" htmlType="submit">
-        Submit
-      </Button>
-    </Form.Item>
-      </Form>
+          <Form form={form} initialValues={course} onFinish={submit}>
+            <h3>
+              Current Invite Code is:&nbsp;
```

Comment:
Instead of `&nbsp;`, can we use normal space?

Reply:
Does not work, the only way I could get a space on the rendered page was with &nbsp